app/classifier/tasks/survey/index.cjsx

```diff
@@ -84,7 +87,13 @@ module.exports = React.createClass
   handleAnnotation: (choice, answers, e) ->
     filters = JSON.parse JSON.stringify @state.filters
     value = @props.annotation.value?.slice?(0) ? []
-    value.push {choice, answers, filters}
+    # @ is undefined within the scope of findIndex
+    currentSelection = @state.selectedChoiceID
+    annotationIndex = value.findIndex (value) -> value.choice is currentSelection
```

Comment:
Oh wait, `selectedChoice` isn't an item in that array, is it.

How about removing the `if` check in favour of
```coffeescript
value = value.filter((value) -> { value.choice !== selectedChoiceID})
value.push({choice, answers, filters)
```

Remove the old answer, if it exists, then add the new one to the end of the array as usual.

So many variables called `value` there 😞